server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/RemoteShardsBalancer.java

```diff
@@ -323,12 +325,16 @@ private void allocateUnassignedReplicas(Queue<RoutingNode> nodeQueue, Map<String
     }
 
     private void ignoreRemainingShards(Map<String, UnassignedIndexShards> unassignedShardMap) {
+        // If any nodes are throttled during allocation, mark all remaining unassigned shards as THROTTLED
+        final UnassignedInfo.AllocationStatus status = anyNodesThrottled
```

Comment:
why are we marking all shards with THROTTLED status if any one node throttled any one shard? the throttling status should be captured at shard level i believe

Reply:
maybe you can ref to Description